app/src/main/AndroidManifest.xml

```diff
@@ -2,14 +2,23 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
           package="de.mathfactory.mooltifill">
 
-    <uses-permission
-            android:name="android.permission.BLUETOOTH"
-            android:maxSdkVersion="30"/>
-    <uses-permission
-            android:name="android.permission.BLUETOOTH_CONNECT"/> <!-- <uses-permission android:name="android.permission.BLUETOOTH_ADMIN" android:maxSdkVersion="30" /> -->
-    <!-- <uses-permission android:name="android.permission.BLUETOOTH_SCAN"/> -->
-    <!-- <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/> -->
-    <!-- <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/> -->
+    <!-- Request legacy Bluetooth permissions on older devices. -->
+    <uses-permission android:name="android.permission.BLUETOOTH" android:maxSdkVersion="30"/>
+
+    <!-- For app to communicate with already-paired Bluetooth devices-->
+    <uses-permission android:name="android.permission.BLUETOOTH_CONNECT"/>
+
+    <!-- BLUETOOTH_ADMIN is for devices older than Android 12 to be able to initiate device discovery -->
+     <uses-permission android:name="android.permission.BLUETOOTH_ADMIN" android:maxSdkVersion="30" />
+
+    <!-- BLUETOOTH_SCAN is needed because app looks for Mooltipass BLE device -->
+     <uses-permission android:name="android.permission.BLUETOOTH_SCAN"/>
+
+    <!-- BLUETOOTH_ADVERTISE is needed for the current device to be discoverable to Mooltipass -->
+    <uses-permission android:name="android.permission.BLUETOOTH_ADVERTISE" />
```

Comment:
BLUETOOTH_ADVERTISE is a runtime permission which, if I'm not mistaken, is not requested in the current form of the app (see also above comment). I suggest do remove this permission. I'm curious: is it possible to do the bonding from the Mooltipass device instead of the Android device? This is what BLUETOOTH_ADVERTISE would be necessary for, right?